File: world-cup-system/src/main/java/com/project/worldcupsystem/models/Team.java
package com.project.worldcupsystem.models;

import javax.persistence.*;
import java.util.List;

@Entity
@Table(name = "team")
public class Team {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "id", nullable = false)
    private Long id;
    private String name;
    private String nationality;
    private Long winsCount;
    private Long drawsCount;
    private Long lossCount;
    private Integer rating;
    @OneToMany
    private List<Match> homeMatches;

    @OneToMany
    private List<Match> awayMatches;

    public Team() {

    }

    public Team(String name, String nationality, Long winsCount, Long drawsCount, Long lossCount, Integer rating, List<Match> homeMatches, List<Match> awayMatches) {
        this.name = name;
        this.nationality = nationality;
        this.winsCount = winsCount;
        this.drawsCount = drawsCount;
        this.lossCount = lossCount;
        this.rating = rating;
        this.homeMatches = homeMatches;
        this.awayMatches = awayMatches;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getNationality() {
        return nationality;
    }

    public void setNationality(String nationality) {
        this.nationality = nationality;
    }

    public Long getWinsCount() {
        return winsCount;
    }

    public void setWinsCount(Long winsCount) {
        this.winsCount = winsCount;
    }

    public Long getDrawsCount() {
        return drawsCount;
    }

    public void setDrawsCount(Long drawsCount) {
        this.drawsCount = drawsCount;
    }

    public Long getLossCount() {
        return lossCount;
    }

    public void setLossCount(Long lossCount) {
        this.lossCount = lossCount;
    }

    public Integer getRating() {
        return rating;
    }

    public void setRating(Integer rating) {
        this.rating = rating;
    }

    public List<Match> getHomeMatches() {
        return homeMatches;
    }

    public void setHomeMatches(List<Match> homeMatches) {
        this.homeMatches = homeMatches;
    }

    public List<Match> getAwayMatches() {
        return awayMatches;
    }

    public void setAwayMatches(List<Match> awayMatches) {
        this.awayMatches = awayMatches;
    }
}

File: world_cup_system_frontend_web/lib/data_models/team.dart
class Team {
  int id;
  String name;
  int winsCount;
  int drawsCount;
  int lossCount;
  int rating;

  Team({
    this.id = 0,
    this.name = '',
    this.winsCount = 0,
    this.drawsCount = 0,
    this.lossCount = 0,
    this.rating = 0,
  });

  static fromJson(jsonResponse) {
    return Team(
      id: jsonResponse['id'],
      name: jsonResponse['name'],
      winsCount: jsonResponse['winsCount'],
      drawsCount: jsonResponse['drawsCount'],
      lossCount: jsonResponse['lossCount'],
      rating: jsonResponse['rating'],
    );
  }

  // convert to json
  Map<String, dynamic> toJson() => {
        'id': id,
        'name': name,
        'winsCount': winsCount,
        'drawsCount': drawsCount,
        'lossCount': lossCount,
        'rating': rating,
      };
}
